### PR TITLE
`config_path()`: return a reference

### PR DESCRIPTION
**Pull request by madsbk**
Avoid a string copy

### CODE REVIEW

**Review by bdice — APPROVED** [review]

File: cpp/include/kvikio/cufile_config.hpp [hunk 1]
```diff
@@ -22,12 +22,12 @@
 namespace kvikio {
 namespace detail {
 
-[[nodiscard]] inline const std::string lookup_config_path()
+[[nodiscard]] inline const char* lookup_config_path()
 {
   const char* env = std::getenv("CUFILE_ENV_PATH_JSON");
   if (env != nullptr && std::filesystem::exists(env)) { return env; }
   if (std::filesystem::exists("/etc/cufile.json")) { return "/etc/cufile.json"; }
```
Comment (vuule) [inline cpp/include/kvikio/cufile_config.hpp:29]: side question: could this be done once (e.g. init a local static var)?

Comment (madsbk) [reply cpp/include/kvikio/cufile_config.hpp:29]: It is done once, `config_path()` calls it to assign a static variable

Comment (vuule) [reply cpp/include/kvikio/cufile_config.hpp:29]: Got it. I assumed repeated calls because of the optimization.

**Pull request merged**